### PR TITLE
Generalize TypeScript declaration

### PR DESCRIPTION
**Pull request by azatoth**
This so we can assign correct signature when importing.

Example:

```ts
import importFresh from "import-fresh";

const path = importFresh<typeof import("path")>("path");
```

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -21,8 +21,11 @@ importFresh('./foo')();
 
 importFresh('./foo')();
 //=> 1
+
+import origFoo = require('./foo');
+const foo = importFresh<origFoo>('./foo')
 ```
 */
-declare function importFresh(moduleId: string): unknown;
+declare function importFresh<T>(moduleId: string): T;
```
Comment (sindresorhus) [inline index.d.ts:29]: There must be a way of reusing the native import type. Something like this (pseudo code):
```suggestion
declare function importFresh(moduleId: string): typeof import(moduleId);
```

File: index.test-d.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
-import {expectType, expectError} from 'tsd';
-import importFresh = require('.');
+import { expectType, expectError } from "tsd";
+import importFresh = require(".");
+import path = require("path");
 
-expectType<unknown>(importFresh('hello'));
+expectType<unknown>(importFresh("hello"));
```
Comment (sindresorhus) [inline index.test-d.ts:5]: Don't do unrelated changes.

File: index.test-d.ts [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
 expectError(importFresh());
+
+expectType<(s: string) => {}>(importFresh<(s: string) => {}>("hello"));
```
Comment (sindresorhus) [inline index.test-d.ts:8]:
```suggestion
expectType<(moduleId: string) => {}>(importFresh<(moduleId: string) => {}>("hello"));
```

File: index.d.ts (hunk 1 above)
Comment (azatoth) [reply index.d.ts:29]: Sadly that doesn't seems to be possible :(

> There must be a way of reusing the native import type. Something like this (pseudo code):

Sadly that doesn't seems to be possible to encode in the typescript declaration :(

Comment (sindresorhus) [reply index.d.ts:28]: We could at least document that `importFresh<typeof import('path')>('path')` works.
